redbpf/src/lib.rs

```diff
@@ -1075,6 +1142,47 @@ impl SocketFilter {
     }
 }
 
+impl SkLookup {
+    /// Attach the `sk_lookup` to the given network namespace.
+    ///
+    /// In most cases it should be attached to `/proc/self/ns/net`.
+    pub fn attach_sk_lookup(&mut self, namespace: &str) -> Result<()> {
+        if self.link.is_some() {
+            return Err(Error::ProgramAlreadyLinked);
+        }
+
+        let fd = self.common.fd.ok_or(Error::ProgramNotLoaded)?;
+        unsafe {
+            let namespace = CString::new(namespace)?;
+            let nfd = libc::open(namespace.as_ptr(), libc::O_RDONLY | libc::O_CLOEXEC);
+            if nfd < 0 {
+                return Err(Error::IO(io::Error::last_os_error()));
+            }
+
+            let lfd = bpf_link_create(fd, nfd, bpf_attach_type_BPF_SK_LOOKUP, ptr::null());
+            if lfd < 0 {
+                libc::close(nfd);
+                return Err(Error::IO(io::Error::last_os_error()));
+            }
+
+            self.link = Some((nfd, lfd));
+        }
+
+        Ok(())
+    }
+}
+
+impl Drop for SkLookup {
+    fn drop(&mut self) {
+        if let Some((nfd, lfd)) = self.link.take() {
+            unsafe {
+                libc::close(lfd);
```

Comment:
@raftario 
I am confused that `bpf_link_detach` is needed or not.
- Only `close` is needed?
- `bpf_link_detach` has nothing to do with cleanup?
- `bpf_link_detach` is called first and then `close` should be called?

Can you explain about that?

Reply:
If you never pin the link, it'll automatically be cleaned up when the file descriptor is closed

Reply:
Oh, I got it. Thank you.